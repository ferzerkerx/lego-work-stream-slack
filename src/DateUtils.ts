export class DateUtils {
  static toPrettyDate(theValue: string): string {
    let theDate: Date = this.parseDate(theValue),
      month = '' + (theDate.getMonth() + 1),
      day = '' + theDate.getDate(),
      year = theDate.getFullYear();

    if (month.length < 2) {
      month = '0' + month;
    }
    if (day.length < 2) {
      day = '0' + day;
    }
    return [year, month, day].join('-');
  }

  static now(): Date {
    const date = new Date();
    date.setHours(0);
    date.setMinutes(0);
    date.setSeconds(0);
    date.setMilliseconds(0);
    return date;
  }

  static add(currentDate: Date, days: number) {
    let date = new Date(currentDate);
    date.setDate(date.getDate() + days);
    return date;
  }

  static parseDate(paramValue: string) {
    const date: Date = new Date(paramValue);
    if (isNaN(date.getTime())) {
      return null;
    }
    return date;
  }
}
