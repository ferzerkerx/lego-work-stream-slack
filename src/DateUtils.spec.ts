import { DateUtils } from './DateUtils';

describe('DateUtils', () => {
  test('toPrettyDate', () => {
    const date = new Date('2018-10-10');
    const dateStr = DateUtils.toPrettyDate(date.toDateString());
    expect(dateStr).toMatchSnapshot();
  });

  test('add', () => {
    const date = new Date('2018-10-10');
    const resultDate = DateUtils.add(date, 2);
    expect(resultDate).toMatchSnapshot();
  });

  test('should parse valid date', () => {
    const resultDate = DateUtils.parseDate('2018-10-10');
    expect(resultDate).toMatchSnapshot();
  });

  test('should not parse invalid date', () => {
    const resultDate = DateUtils.parseDate('not a date');
    expect(resultDate).toBeNull();
  });
});
