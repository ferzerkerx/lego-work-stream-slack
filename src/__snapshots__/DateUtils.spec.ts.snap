// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateUtils add 1`] = `2018-10-12T00:00:00.000Z`;

exports[`DateUtils should parse valid date 1`] = `2018-10-10T00:00:00.000Z`;

exports[`DateUtils toPrettyDate 1`] = `"2018-10-10"`;
